### DISCUSSION TITLE
Lighthouse deck not detected

### DISCUSSION THREAD

**Discussion by FredrikHolsten** [bitcraze/discussions#1207] [General]
Hi!
First of all, thanks for creating an awesome product!

I have a lighthouse setup and everything is working well for all but two of my cf2.1's.
The first one starts up and the self-test passes, but the blue and red light on the lighthouse deck is constantly lit and it seems that the deck isn't recognized:

SYS: Crazyflie 2.1 is up and running!
SYS: Build 0:e527895e8c6e (2023.11) CLEAN
SYS: I am 0x333733363432470500370020 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_INFO: Memory error: wrong header ID
DECK_INFO: Deck 0 has corrupt OW memory. No driver will be initialized!
DECK_CORE: 0 deck(s) found
IMU: BMI088: Using I2C interface.
IMU: BMI088 Gyro connection [OK].
IMU: BMI088 Accel connection [OK]
IMU: BMP388 I2C connection [OK]
ESTIMATOR: Using Complementary (1) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
SYS: About to run tests in system.c.
SYS: NRF51 version: 2024.2 (CF21)
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
IMU: BMI088 gyro self-test [OK]
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 20496 bytes
STAB: Starting stabilizer loop
SUP: Can not fly
SUP: Ready to fly

The second crazyflie does not even run the self test when the deck is on. No lights are lit on the deck and only the two blue lights are lit on the crazyflie. 

I would really appreciate any help.

**Comment by ataffanel** [comment]
This looks like connections problems. Could you check that the deck connectors on your Crazyflies looks good, that the metal in the connector is doing good contact with the pins. You can also try to swap the pins.

Do you have any other decks to test? It would be interesting to know if the problem actually comes from these Crazyflies or if it is the lighthouse deck that is acting randomly.

**Reply by FredrikHolsten** [reply]
Thanks for your reply. 
I have tested with other lighthouse decks and its the same problem.
I have to agree that it seems to be an issue with the connectors on the crazyflies, although I'm not sure if I understand how to check if the connectors are in good contact with the pins.
Would you elaborate? :)

**Reply by tobbeanton** [reply]
There are some pointers in [this](https://github.com/orgs/bitcraze/discussions/409) discussion.

**Reply by skovsen** [reply]
Hi. I'm a colleague of @FredrikHolsten and I have the CFs in my possession.
The first CF, witch has the log first posted, shows both the red and blue LED lighted up.
I have also tried to write new firmware on it, using the CFClient. During that process it discovers a deck (the LH deck is the only connected) and it outputs that new FW is written to that as well, but it I still get the same log message when trying to connect to the CF. 
I have tried with another LH deck and that works fine, so it is definitely that deck. I can see in the log that it mentions that the OW memory is corrupt, is there a way to flash this or what can we do?

**Reply by tobbeanton** [reply]
It could be the connection to the OW memory that is glitch since this "should" never get corrupted without actively writing it. You can try to rewrite it though:

1. Place the unpacked [write-ow-lh.zip](https://github.com/bitcraze/discussions/files/14720865/write-ow-lh.zip)  in the crazyflie-lib-python/examples/memory folder 
2. Run it with "python3 write-ow-lh.py" with only one CF2.1 powered on, a crazyradio and only the lighthouse deck attached (it will connect to the first CF it finds)
3. Hopefully it succeds so then reboot
4. Connect and is should now work

**Reply by skovsen** [reply]
@tobbeanton Thanks for the help. I got one of the boards up and working again.

I still have one LH board that is not working. When I start the CF with it on, it just ends up with solid blue light in M2 and M3 and no self-test and no way of connecting to it. Do you know what that might mean?

Regards

**Reply by tobbeanton** [reply]
It sounds like the CF gets stuck in the boot sequence for some reason. Most likely because of some electrical problem. If you have pin headers to spare you could start by "pulling away" all but the GND, VCOM pins. The CF should then start. Then add one pin after the other until the problem appears.
